hazelcast/src/main/java/com/hazelcast/cache/ICache.java

```diff
@@ -944,4 +945,31 @@
      */
     CacheStatistics getLocalCacheStatistics();
 
+    /**
+     * Adds a CachePartitionLostListener.
+     * <p/>
+     * The addPartitionLostListener returns a register-id. This id is needed to remove the MapPartitionLostListener using the
+     * {@link #removePartitionLostListener(String)} method.
+     * <p/>
+     * There is no check for duplicate registrations, so if you register the listener twice, it will get events twice.
+     * IMPORTANT: Please @see com.hazelcast.partition.PartitionLostListener for weaknesses.
+     * IMPORTANT: Listeners registered from HazelcastClient may miss some of the cache partition lost events due
+     * to design limitations.
+     *
+     * @param listener the added MapPartitionLostListener.
+     * @return returns the registration id for the MapPartitionLostListener.
+     * @throws java.lang.NullPointerException if listener is null.
+     * @see #removePartitionLostListener(String)
+     */
+    String addPartitionLostListener(CachePartitionLostListener listener);
+
+    /**
+     * Removes the specified cache partition lost listener.
+     * Returns silently if there is no such listener added before.
+     *
+     * @param id id of registered listener.
+     * @return true if registration is removed, false otherwise.
+     */
+    boolean removePartitionLostListener(String id);
```

Comment:
you are throwing NPE when id is `null`.. Can you also write this info to javadoc?